ee/app/livechat-enterprise/server/methods/resumeOnHold.ts

```diff
@@ -0,0 +1,52 @@
+import { Meteor } from 'meteor/meteor';
+
+import { Subscriptions, LivechatRooms, LivechatInquiry } from '../../../../../app/models/server';
+import { callbacks } from '../../../../../app/callbacks/server';
+import { RoutingManager } from '../../../../../app/livechat/server/lib/RoutingManager';
+import { Livechat } from '../../../../../app/livechat/server';
+import { dispatchInquiryPosition } from '../lib/Helper';
+
+
+Meteor.methods({
+	async 'livechat:resumeOnHold'(roomId, options = { clientAction: false }) {
+		const room = await LivechatRooms.findOneById(roomId);
+		if (!room || room.t !== 'l') {
+			throw new Meteor.Error('error-invalid-room', 'Invalid room', { method: 'livechat:resumeOnHold' });
+		}
+
+		if (!room.onHold) {
+			throw new Meteor.Error('room-closed', 'Room is not OnHold', { method: 'livechat:resumeOnHold' });
+		}
+
+		const { servedBy: { _id: agentId = null, username = null } = {} } = room;
+		let agent: any = { agentId, username };
+		const inquiry = await LivechatInquiry.findOneByRoomId(roomId, {});
+		const { departmentId } = inquiry;
```

Comment:
All methods below aren't supposed to be here.
As mentioned before, the `QueueManager` is responsible for dealing with `inquiries` and even routing them.

It's not a good practice to spread those methods over the codebase.

Reply:
Okay got it. will keep this in mind.
Moved this logic inside Queue Manager in this commit -> https://github.com/RocketChat/Rocket.Chat/pull/20945/commits/bf2d641fa478c1bc1267361077877d392ab7d409